document_ai_warehouse/batch-ingestion-docai-wh/load_docs.py

```diff
@@ -0,0 +1,493 @@
+    reference_id: str,
+    document_ai_output,
+    metadata_properties: List[contentwarehouse_v1.Property],
+):
+    create_document_response = dw_utils.create_document(
+        display_name=os.path.basename(file_uri),
+        mime_type="application/pdf",
+        document_schema_id=document_schema_id,
+        raw_document_path=file_uri,
+        docai_document=document_ai_output,
+        caller_user_id=CALLER_USER,
+        reference_id=reference_id,
+        metadata_properties=metadata_properties,
+    )
+
+    Logger.debug(
+        f"create_document_response={create_document_response}"
+    )  # Verify that the properties have been set correctly
+
+    if create_document_response is not None:
+        document_id = create_document_response.document.name.split("/")[-1]
+        document_id_list.append(document_id)
+
+        dw_utils.link_document_to_folder(
+            document_id=document_id,
+            folder_document_id=folder_id,
+            caller_user_id=CALLER_USER,
+        )
+        Logger.info(
+            f"Created document {file_uri} with reference_id={reference_id} inside folder_id={folder_id} and using schema_id={document_schema_id}"
+        )
+        return document_id
+
+
+def create_folder_schema(schema_path: str):
+    folder_schema = storage_utils.read_file(schema_path, mode="r")
+    display_name = json.loads(folder_schema).get("display_name")
+
+    for ds in dw_utils.list_document_schemas():
+        if ds.display_name == display_name and ds.document_is_folder:
+            return ds.name.split("/")[-1]
+
+    create_schema_response = dw_utils.create_document_schema(folder_schema)
+    folder_schema_id = create_schema_response.name.split("/")[-1]
+
+    Logger.info(f"folder_schema_id={folder_schema_id}")
+    response = dw_utils.get_document_schema(schema_id=folder_schema_id)
+    Logger.debug(f"response={response}")
+    return folder_schema_id
+
+
+def create_folder(folder_schema_id: str, display_name: str, reference_id: str):
+    reference_path = f"referenceId/{reference_id}"
+    try:
+        document = dw_utils.get_document(reference_path, CALLER_USER)
+        folder_id = document.name.split("/")[-1]
+        return folder_id
+    except NotFound:
+        Logger.info(
+            f" -------> Creating sub-folder [{display_name}] with reference_id=[{reference_id}]"
+        )
+        create_folder_response = dw_utils.create_document(
+            display_name=display_name,
+            document_schema_id=folder_schema_id,
+            caller_user_id=CALLER_USER,
+            reference_id=reference_id,
+        )
+        if create_folder_response is not None:
+            folder_id = create_folder_response.document.name.split("/")[-1]
+            return folder_id
+
+
+def get_document_schemas():
+    schemas = {}
+    for ds in dw_utils.list_document_schemas():
+        if ds.display_name not in schemas:
+            schemas[ds.display_name] = ds.name.split("/")[-1]
+    return schemas
```

Comment:
```suggestion
def get_document_schemas():
    return {ds.display_name: ds.name.split("/")[-1] for ds in dw_utils.list_document_schemas()}
```

More Pythonic, although I also recommend swapping out the `split()` for the client library function that gets the document id or whatever you're accessing.